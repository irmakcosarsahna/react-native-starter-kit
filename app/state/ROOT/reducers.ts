export * from '../common/reducer';
export * from '../auth/reducers';
export * from '../theme/reducers';
