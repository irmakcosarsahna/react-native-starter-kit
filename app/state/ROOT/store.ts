import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { persistReducer, persistStore } from 'redux-persist'; // defaults to localStorage for web
import rootSaga from './middleware/saga';
import * as reducers from './reducers';
import { common as commonPersistConfig, root as rootPersistConfig } from './config';
import { successGenerate } from '../common';
import * as Types from '../auth/user/types';

const sagaMiddleware = createSagaMiddleware();
const middlewares = [applyMiddleware(sagaMiddleware)];
// reducer
const appReducer = combineReducers({
  ...reducers,
  common: persistReducer(commonPersistConfig, reducers.common),
});
// Logout
const rootReducer = (state, action) => {
  if (action.type === successGenerate(Types.LOGOUT)) {
    state = undefined;
  }
  return appReducer(state, action);
};
// persist
const pReducer = persistReducer(rootPersistConfig, rootReducer);

// store
const store = createStore(
  pReducer, // root reducer with router state
  compose(...middlewares),
);

// Go go go
sagaMiddleware.run(rootSaga);

const persistor = persistStore(store);

export { persistor, store };
