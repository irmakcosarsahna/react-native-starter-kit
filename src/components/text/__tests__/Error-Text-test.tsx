/**
 * @format
 */

import { store } from '@state/ROOT';
import React from 'react';
import 'react-native';
import { Provider } from 'react-redux';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { Text } from '../index';

it('renders Error Text', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Text text={'error test'} />
    </Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
