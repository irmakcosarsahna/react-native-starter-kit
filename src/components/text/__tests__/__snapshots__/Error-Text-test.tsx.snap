// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Error Text 1`] = `
<Text
  allowFontScaling={false}
  style={
    Object {
      "color": "#1A1A1A",
      "fontFamily": "Arial",
      "fontSize": 15,
    }
  }
>
  error test
</Text>
`;
